realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -605,6 +620,9 @@ public Builder rxFactory(RxObservableFactory factory) {
          * @param transaction transaction to execute.
          */
         public Builder initialData(Realm.Transaction transaction) {
+            if (readOnly) {
```

Comment:
Check and throw in the `Builder.build()` instead. Not worth to check here plus this doesn't deal with `initialData()` first then `readOnly()`.